src/model/trace-viewer.js

```diff
@@ -13,7 +13,7 @@
 // limitations under the License.
 
 // eslint-disable-next-line import/prefer-default-export
-export function getTraceName(spans, processes) {
-  const span = spans.find(sp => sp.spanID === sp.traceID) || spans[0];
-  return span ? `${processes[span.processID].serviceName}: ${span.operationName}` : '';
+export function getTraceName(spans) {
+  const span = spans.filter(sp => !sp.references || !sp.references.length)[0];
```

Comment:
it is possible that the root span is delayed en-route to the storage when the UI tries to show the trace, so `[0]` might try to dereference an empty array. I'd rather not assume it exists.

Reply:
Dereference an empty array in JS does not error, it returns `undefined`; this is a safe operation. The `.filter()` method will always return an Array.